### PR TITLE
feat: add pgvector as vectorstore

### LINKED ISSUE #122: Support pgvector
pgvector : https://github.com/pgvector/pgvector is open-source vector similarity search for Postgres

which means you should implement a vectorstore in langchaingo https://github.com/tmc/langchaingo/blob/main/vectorstores/vectorstores.go#L12 

You can host code under `kubeagi/arcadia pkg/vectorstores/pgvector`

### PR DESCRIPTION
**Pull request by Abirdcfly**
#### What type of PR is this?
/kind feature

#### Which issue(s) this PR fixes

Fixes #122 

#### Special notes for your reviewer

<table>
    <tr>
        <th>collection_id</th>
        <th>embedding</th>
        <th>document</th>
        <th>cmetadata</th>
        <th>custom_id</th>
        <th>uuid</th>
    </tr>
    <tr>
        <td>a79c2a40-f294-4f41-9329-dfbb7587ce42</td>
        <td>&#91;-0.0012137949&#44;-0.01627453&#44;-0.29887754&#44;-0.15064785&#44;0.18556964&#44;-0.0054863095&#44;-0.018341854&#44;0.14186022&#44;-0.15160587&#44;0.32042432&#44;0.12986974&#44;0.10587247&#44;0.062361706&#44;0.19167428&#44;-0.122019455&#44;-0.19534577&#44;0.18896861&#44;0.074173495&#44;-0.20890827&#44;-0.114406824&#44;0.058532856&#44;-0.13274224&#44;-0.03907444&#44;-0.027943974&#44;0.10992267&#44;0.11519617&#44;-0.12559476&#44;-0.13631818&#44;-0.096149966&#44;-0.28476387&#44;-0.036539335&#44;0.07999022&#44;0.10625024&#44;-0.081889704&#44;-0.012332646&#44;0.0511827&#44;-0.30168033&#44;0.25154486&#44;0.22629686&#44;-0.07815949&#44;-0.012347159&#44;0.30410376&#44;0.14833578&#44;-0.057312656&#44;-0.0022729754&#44;0.062316865&#44;0.115179636&#44;0.078375235&#44;-0.019557651&#44;0.13999477&#44;0.0026726245&#44;-0.0486658&#44;-0.142802&#44;-0.13045342&#44;-0.24166664&#44;-0.24624738&#44;-0.16341826&#44;0.09350339&#44;0.18473077&#44;0.4284048&#44;-0.06657189&#44;0.32504377&#44;0.064574026&#44;0.13750733&#44;-0.099156424&#44;-0.02423546&#44;0.11913091&#44;0.00695329&#44;-0.24502592&#44;-0.27141935&#44;-0.012099944&#44;0.032004178&#44;-0.24789572&#44;-0.04784686&#44;0.17778829&#44;-0.07966299&#44;0.16059065&#44;-0.1467757&#44;0.19080648&#44;0.21836963&#44;0.20416877&#44;0.2002398&#44;-0.09443957&#44;0.050842285&#44;-0.1425586&#44;0.08621045&#44;-0.0014072836&#44;0.15081774&#44;0.2979713&#44;-0.054784674&#44;-0.12767677&#44;0.33520865&#44;0.06613692&#44;-0.058446027&#44;-0.011237973&#44;0.105889365&#44;0.16255923&#44;-0.15268524&#44;-0.17803614&#44;-0.13315453&#44;0.053300798&#44;-0.053506482&#44;-0.12538627&#44;-0.07583292&#44;-0.018486986&#44;0.35445178&#44;0.05397554&#44;-0.029658651&#44;0.16016385&#44;0.047108337&#44;-0.043380104&#44;0.2796821&#44;-0.04973564&#44;-0.08290937&#44;0.04997021&#44;0.060030054&#44;0.02208012&#44;-0.07208617&#44;0.20205872&#44;-0.33760506&#44;-0.027052347&#44;-0.011205064&#44;0.09237504&#44;0.0014191329&#44;-0.03541252&#44;-0.04602454&#44;0.30219704&#44;-0.20137262&#44;-0.09911149&#44;-0.08688988&#44;0.04972934&#44;-0.17415316&#44;-0.24589154&#44;0.029692184&#44;-0.26894826&#44;0.12542208&#44;-0.16370349&#44;0.033127576&#44;-0.22902513&#44;0.12925275&#44;0.03842619&#44;0.06592233&#44;0.108465746&#44;0.366819&#44;0.0590656&#44;-0.36322707&#44;0.054757185&#44;0.047969557&#44;0.14267996&#44;0.19340834&#44;0.2779137&#44;0.21363243&#44;-0.021875393&#44;-0.21688259&#44;0.19810452&#44;0.09590119&#44;-0.29552466&#44;-0.092654504&#44;0.09103398&#44;-0.13666482&#44;0.07758307&#44;0.106437005&#44;0.024839401&#44;0.013622075&#44;-0.25166267&#44;0.07433418&#44;-0.030963516&#44;-0.35626417&#44;0.07873474&#44;-0.34017643&#44;0.12742274&#44;0.2321529&#44;0.029199917&#44;-0.2307467&#44;-0.20127635&#44;-0.003339513&#44;-0.014847132&#44;0.0067399233&#44;0.012364805&#44;0.021094654&#44;0.21676917&#44;-0.32497483&#44;0.079431176&#44;0.16660671&#44;0.19502953&#44;0.016223159&#44;0.0071525513&#44;0.032592498&#44;-0.15752788&#44;-0.01890325&#44;-0.062320806&#44;-0.36790416&#44;0.044657104&#44;0.092752546&#44;0.23554568&#44;0.24326412&#44;-0.1263734&#44;-0.2635668&#44;-0.17287974&#44;0.13353108&#44;0.082927406&#44;-0.20786023&#44;0.08502566&#44;-0.031721868&#44;-0.13940695&#44;0.011748376&#44;0.11471371&#44;-0.13394335&#44;0.057450026&#44;0.32656735&#44;-0.20116258&#44;-0.08884704&#44;0.017516539&#44;-0.17212987&#44;-0.08419243&#44;0.16023593&#44;0.17248526&#44;0.35698545&#44;-0.02530522&#44;0.114697434&#44;-0.38529307&#44;-0.02747267&#44;0.095379435&#44;-0.030384678&#44;-0.067339495&#44;-0.20595868&#44;-0.059295572&#44;-0.045949113&#44;0.077019274&#44;-0.06557373&#44;-0.21299405&#44;0.11277471&#44;-0.0018990592&#44;0.116786346&#44;-0.038882762&#44;-0.023051104&#44;0.08063882&#44;0.31854373&#44;0.081830405&#44;0.06540259&#44;-0.09817882&#44;-0.22638173&#44;-0.2766449&#44;0.076662675&#44;0.29853&#44;-0.11754377&#44;-0.11114283&#44;0.020340208&#44;-0.03874865&#44;-0.045976363&#44;-0.22232647&#44;0.09774196&#44;-0.19859213&#44;-0.20114632&#44;-0.28875536&#44;0.010973337&#44;-0.15269789&#44;-0.101234235&#44;-0.05759543&#44;-0.26849586&#44;0.18741485&#44;0.07826705&#44;0.36285567&#44;-0.1694862&#44;0.10275892&#44;0.115667&#44;0.08599068&#44;-0.124871776&#44;0.039363477&#44;0.10798719&#44;-0.08584738&#44;-0.135667&#44;-0.14323823&#44;0.012743945&#44;0.07137223&#44;0.3434019&#44;-0.16571918&#44;-0.015593651&#44;0.14362608&#44;-0.09795326&#44;-0.15382405&#44;-0.09223075&#44;0.0898623&#44;0.23007254&#44;-0.12909117&#44;-0.21103902&#44;0.10210954&#44;-0.22464141&#44;-0.08775505&#44;0.22562127&#44;0.07097672&#44;-0.18151513&#44;0.16695444&#44;-0.089870945&#44;-0.38296142&#44;0.12426078&#44;-0.24274679&#44;-0.13657501&#44;-0.080066346&#44;0.114820585&#44;-0.026254814&#44;0.12708475&#44;-0.07946059&#44;-0.078123316&#44;-0.033568762&#44;0.02819525&#44;-0.045604866&#44;0.07869432&#44;0.15290047&#44;0.1297495&#44;0.03769469&#44;0.040938795&#44;-0.028377812&#44;0.08166784&#44;0.13124733&#44;-0.17225532&#44;-0.10505494&#44;-0.13580497&#44;0.025383735&#44;-0.14786895&#44;-0.2760997&#44;0.31993487&#44;0.11639042&#44;-0.010311494&#44;0.01284188&#44;-0.16434388&#44;0.011988032&#44;-0.13174647&#44;-0.11749931&#44;-0.13822027&#44;0.04663176&#44;0.13450964&#44;-0.053085715&#44;0.00026913063&#44;0.1535886&#44;-0.1735427&#44;0.1049483&#44;-0.045983654&#44;-0.13808817&#44;-0.19282928&#44;0.22031288&#44;-0.21674907&#44;-0.010833937&#44;-0.033284098&#44;0.11744124&#44;0.12786579&#44;-0.3145692&#44;0.05414653&#44;-0.08739773&#44;0.11737295&#44;0.32660708&#44;-0.024006376&#44;-0.010516489&#44;0.19013275&#44;-0.032883912&#44;0.19758609&#44;-0.12055139&#44;0.02808721&#44;-0.040796943&#44;0.06787499&#44;0.121222615&#44;-0.15359206&#44;-0.09140733&#44;-0.11350989&#44;-0.010341877&#44;0.1575319&#44;0.02610753&#44;0.22167566&#44;0.090685174&#44;0.03804333&#44;0.037979733&#44;0.05742029&#44;-0.033608675&#44;-0.0084355455&#44;-0.046110567&#44;-0.056862764&#44;0.037491415&#44;0.23342879&#44;0.1460526&#44;0.016794842&#44;0.04038862&#44;-0.08757304&#44;0.023402834&#44;0.07001301&#44;0.07408565&#44;-0.023577165&#44;0.0036034673&#44;-0.16102715&#44;0.018971456&#44;0.22069617&#44;-0.16070214&#44;0.08103865&#44;-0.09438695&#44;-0.19222975&#44;-0.10494866&#44;-0.0011375204&#44;-0.09431766&#44;-0.29002458&#44;0.22614431&#44;-0.11916206&#44;-0.20667568&#44;-0.15737103&#44;-0.11038387&#44;-0.033033054&#44;0.2748346&#44;-0.21248627&#44;0.06435461&#44;0.15322971&#44;-0.0409408&#44;0.14914581&#44;-0.03258024&#44;0.0437159&#44;0.0004085841&#44;0.14304978&#44;0.08841152&#44;0.21691473&#44;-0.15346165&#44;-0.16894004&#44;0.06571084&#44;0.075187124&#44;0.17904958&#44;0.18569154&#44;-0.22827646&#44;-0.112693444&#44;-0.049323995&#44;0.04476766&#44;-0.15157442&#44;0.007886285&#44;0.13137542&#44;0.16959342&#44;0.019437332&#44;0.3034669&#44;-0.21984538&#44;0.046143025&#44;-0.05200086&#44;0.22618024&#44;-0.12811515&#44;-0.04395021&#44;0.018662851&#44;0.2666656&#44;0.055861842&#44;-0.07550811&#44;-0.13424301&#44;-0.044374038&#44;-0.09816122&#44;-0.0378616&#44;0.026718572&#44;0.28944704&#44;-0.011605704&#44;-0.078224674&#44;0.09135046&#44;-0.015131461&#44;0.14277944&#44;-0.124198854&#44;-0.1264849&#44;-0.1916459&#44;-0.20634827&#44;-0.14875652&#44;-0.023308074&#44;-0.07814376&#44;0.076192774&#44;-0.043312706&#44;0.2975331&#44;-0.063550815&#44;-0.0781299&#44;-0.14858826&#44;0.012995395&#44;-0.04056874&#44;0.21839842&#44;-0.038585477&#44;-0.15336837&#44;-0.1976771&#44;-0.003640391&#44;-0.119367026&#44;0.009487239&#44;0.055576872&#44;-0.11735777&#44;0.28139457&#44;0.36074105&#44;-0.40528947&#44;0.26464868&#44;0.028161779&#44;0.01668623&#44;0.13530973&#44;0.032194488&#44;0.13331492&#44;-0.12788332&#44;0.00012236834&#44;-0.112648726&#44;0.15321861&#44;0.06924368&#44;-0.017459398&#44;0.03303682&#44;0.042544913&#44;-0.11520469&#44;-0.24368283&#44;0.052214693&#44;-0.017674807&#44;0.11536841&#44;-0.101808034&#44;-0.10408932&#44;0.06251663&#44;0.09485161&#44;-0.12621412&#44;-0.28671166&#44;0.31958708&#44;0.26776093&#44;-0.08657151&#44;-0.1294876&#44;-0.22142084&#44;0.04900673&#44;0.09830935&#44;0.14479086&#44;-0.2155428&#44;-0.08517798&#44;0.094546504&#44;-0.14710455&#44;0.09655792&#44;0.03727327&#44;-0.07315047&#44;-0.15774477&#44;-0.11846864&#44;-0.1614166&#44;0.19353452&#44;-0.12815225&#44;-0.16625981&#44;0.34824443&#44;-0.08587541&#44;-0.13404691&#44;-0.01433638&#44;0.083703734&#44;-0.08646386&#44;0.28575045&#44;0.12313546&#44;-0.039627224&#44;0.22919004&#44;0.13194111&#44;-0.00315713&#44;-0.18475394&#44;-0.14891835&#44;0.074332416&#44;0.031858504&#44;0.10284189&#44;0.21005452&#44;-0.13111202&#44;0.18879102&#44;-0.1453358&#44;-0.1233956&#44;-0.18688017&#44;-0.17172524&#44;0.087689206&#44;-0.18815859&#44;-0.047631096&#44;0.16016446&#44;0.16857186&#44;0.06972718&#44;0.341668&#44;0.10982921&#44;-0.22060609&#44;0.06156236&#44;-0.16898294&#44;-0.09831973&#44;0.038960908&#44;0.03957661&#44;-0.23300305&#44;-0.047734033&#44;-0.13326517&#44;0.22050431&#44;0.19212823&#44;-0.03787114&#44;0.021120742&#44;0.24966268&#44;0.016220324&#44;-0.13309206&#44;0.13722755&#44;-0.06215682&#44;0.15012826&#44;-0.056153554&#44;0.108525515&#44;0.1260101&#44;-0.120458424&#44;-0.056174535&#44;-0.08354898&#44;0.14590257&#44;0.0756943&#44;0.17720458&#44;0.12953606&#44;0.07712829&#44;-0.09195644&#44;0.025114555&#44;-0.14688392&#44;-0.11574571&#44;0.004936653&#44;-0.15788214&#44;-0.11354645&#44;0.021774217&#44;0.22315279&#44;0.16758734&#44;-0.19199035&#44;0.24984542&#44;-0.14508727&#44;-0.015182987&#44;0.03488981&#44;-0.1986157&#44;0.18567285&#44;0.0016622662&#44;0.08468509&#44;0.05601251&#44;-0.30027384&#44;0.029856969&#44;0.04098266&#44;0.118859425&#44;-0.1217386&#44;0.049351793&#44;-0.14092459&#44;-0.18378383&#44;0.13530675&#44;0.22543104&#44;-0.12047567&#44;-0.2150704&#44;0.050391722&#44;-0.21910301&#44;-0.06620353&#44;-0.13465199&#44;0.08723872&#44;0.07143836&#44;-0.061856292&#44;0.059810985&#44;0.24955764&#44;0.15004578&#44;-0.1489398&#44;-0.028155213&#44;-0.02591927&#44;-0.19586173&#44;0.07016875&#44;0.022824585&#44;0.14966114&#44;-0.03360662&#44;0.14284138&#44;0.1713458&#44;-0.13050325&#44;0.04056429&#44;-0.12979981&#44;-0.1972702&#44;0.02269763&#44;0.021324387&#44;-0.27750868&#44;-0.08088839&#44;0.08631213&#44;-0.05408119&#44;0.16004011&#44;-0.16166274&#44;0.2091649&#44;-0.08543437&#44;0.19403635&#44;0.39749315&#44;0.029443884&#44;0.16504829&#44;-0.14703006&#44;0.009466055&#44;0.2342668&#44;0.053985715&#44;-0.08539684&#44;-0.25745425&#44;-0.1159595&#44;-0.19106126&#44;0.18172005&#44;0.23501906&#44;-0.215023&#44;-0.323414&#44;0.32340354&#44;-0.2792732&#44;-0.2567249&#44;-0.019907331&#44;0.03823605&#44;-0.38382795&#44;-0.14623606&#44;0.02747103&#44;-0.347159&#44;0.010028121&#44;0.06760897&#44;0.13632074&#44;0.06340208&#44;0.14206883&#44;0.2345753&#44;0.080727145&#44;-0.15564673&#44;-0.024865095&#44;-0.08898155&#44;0.031218445&#44;0.09585983&#44;6.373153e-05&#44;-0.1637986&#44;0.30124307&#44;-0.011609349&#44;0.056100834&#44;-0.1937637&#44;-0.16387022&#44;0.39119545&#44;0.30788156&#44;0.16239011&#44;-0.18149185&#44;-0.080758065&#44;-0.12412661&#44;0.021951173&#44;-0.28123343&#44;0.18607137&#44;-0.14966851&#44;0.10303108&#44;-0.0024485625&#44;0.21423975&#44;-0.0057421983&#44;0.4328487&#44;0.012095576&#44;-0.035211556&#44;0.29022866&#44;-0.03798087&#44;0.1984971&#44;-0.105520055&#44;-0.2522771&#44;-0.19096348&#44;-0.4750665&#44;0.16674903&#44;0.091776274&#44;0.051136963&#44;-0.2122499&#44;-0.150732&#44;0.32473555&#44;0.06284635&#44;0.08699967&#44;0.12265225&#44;0.12865898&#44;0.16183023&#44;0.081473485&#44;0.0065360083&#44;-0.0808399&#44;0.12219375&#44;-0.047004193&#44;-0.008061928&#44;-0.042328034&#44;-0.0065724356&#44;0.11471844&#44;0.035449058&#44;0.04170733&#44;0.061031353&#44;0.07106969&#44;0.3383735&#44;0.1318867&#44;-0.2556902&#44;-0.3076729&#44;-0.044307195&#44;0.19105208&#44;0.028545931&#44;0.13342485&#44;-0.1358627&#44;0.18742475&#44;-0.028500315&#44;-0.1497369&#44;0.28714794&#44;0.18945235&#44;-0.10645827&#44;0.00517585&#44;0.06587036&#44;-0.04183644&#44;0.03882249&#44;-0.25685543&#44;-0.018529631&#44;-0.08309035&#44;-0.14309162&#44;-0.05948738&#44;0.004470716&#44;0.041996438&#44;-0.022123957&#44;0.1931319&#44;-0.023905132&#44;-0.05986593&#44;0.10936747&#44;0.11848509&#44;-0.0047284663&#44;-0.12214198&#44;0.15904304&#44;-0.42881694&#44;-0.023537135&#44;0.018674273&#44;-0.4506051&#44;-0.072636224&#44;0.10764074&#44;-0.06893481&#44;-0.05931682&#44;0.07690036&#44;0.0051036654&#44;0.13066404&#44;0.23117967&#44;0.1271193&#44;-0.09848816&#44;0.10070238&#44;0.10268321&#44;0.069573835&#44;0.07129276&#44;0.15142234&#44;2.6952488e-08&#44;-0.17795818&#44;-0.092227444&#44;0.010382788&#44;0.099097416&#44;-0.13359335&#44;-0.016316168&#44;-0.054000378&#44;0.03692802&#44;0.19407253&#44;-0.122464195&#44;-0.12985182&#44;-0.10343224&#44;0.027980229&#44;-0.02925679&#44;-0.069459036&#44;0.26077658&#44;0.2325761&#44;-0.22494979&#44;-0.35052785&#44;-0.18488026&#44;0.016772753&#44;-0.16411692&#44;-0.06401871&#44;0.29066893&#44;0.046266332&#44;0.17858358&#44;-0.08507654&#44;0.4093646&#44;0.021241898&#44;0.01666185&#44;-0.11904751&#44;0.18386474&#44;0.18999591&#44;0.23513703&#44;-0.2918889&#44;-0.048257437&#44;0.15092187&#44;-0.13608257&#44;-0.07820374&#44;-0.08509189&#44;0.025134867&#44;-0.20177901&#44;0.07099413&#44;-0.0005590923&#44;0.030711407&#44;0.13437945&#44;-0.2766088&#44;0.13121772&#44;0.13664015&#44;0.04606528&#44;0.122526005&#44;-0.36132658&#44;-0.033400647&#44;-0.23706226&#44;0.17108075&#44;0.050895117&#44;-0.006458399&#44;0.2846263&#44;-0.20494004&#44;0.101886556&#44;-0.010426725&#44;0.3907953&#44;0.026768267&#44;-0.023191502&#44;-0.16400321&#44;0.19998226&#44;0.3281369&#44;0.10478467&#44;-0.25703892&#44;-0.25895566&#44;-0.24948911&#44;0.025767332&#44;0.009872859&#44;0.4012494&#44;0.21715176&#44;-0.12517363&#44;-0.10176875&#44;0.1430722&#44;0.008418471&#44;0.1046566&#44;-0.16841504&#44;-0.123989485&#44;-0.12723605&#44;-0.009732494&#44;0.009725938&#44;-0.33919978&#44;0.31841502&#44;-0.18722284&#44;0.13420987&#44;-0.079814&#44;0.04116376&#44;-0.100009605&#44;0.12049477&#44;0.22349708&#44;0.16837278&#44;-0.035339296&#44;-0.30119655&#44;-0.1242799&#44;0.29853153&#44;-0.036236234&#44;-0.07938617&#44;0.11440082&#44;-0.049718417&#44;-0.04456926&#44;0.20100817&#44;-0.007291584&#44;-0.020609934&#44;0.15228572&#44;0.11598585&#44;-0.15913704&#44;-0.12522918&#44;-0.17781723&#44;-0.060534786&#44;0.31549016&#44;-0.30503604&#44;0.13317318&#44;0.13611548&#44;-0.008808553&#44;0.18597165&#44;-0.18858114&#44;0.03518981&#44;0.31020394&#44;-0.13543317&#44;-0.078912005&#44;-0.08679686&#44;0.17449239&#44;-0.09263186&#44;-0.010966027&#44;0.13572778&#44;0.15571497&#44;0.15192756&#44;0.11532233&#44;-0.013091694&#44;0.15492032&#44;0.106454395&#44;-0.063637495&#44;-0.22009988&#44;0.028187012&#44;0.049077738&#44;0.12239257&#44;0.5830882&#44;0.28530228&#44;0.0758682&#44;-0.4135266&#44;-0.14716871&#44;0.09000088&#44;-0.19009797&#44;0.20973916&#44;0.014257302&#44;-0.08131992&#44;-0.117750145&#44;-0.27979678&#44;-0.053777557&#44;0.018542215&#44;0.11327298&#44;-0.1382599&#44;-0.060703795&#44;-0.053437077&#44;-0.14069197&#44;-0.3001089&#44;0.038753945&#44;-0.15088207&#44;0.08383961&#44;-0.008418026&#44;0.18105833&#44;-0.22659521&#44;0.29692104&#44;0.39215142&#44;-0.012091652&#44;0.17343655&#44;-0.21304849&#44;-0.09548889&#44;-0.020550843&#44;0.14315537&#44;0.21086541&#44;0.046838112&#44;0.02191762&#44;0.023431713&#44;0.20443794&#44;-0.08538741&#44;0.02601314&#44;-0.16972753&#44;0.11462836&#44;-0.11805041&#44;0.39604622&#44;-0.025860485&#44;0.05184076&#44;0.03356752&#44;-0.045890413&#44;0.19937256&#44;0.04414761&#44;-0.06486678&#44;0.17000762&#44;0.20330913&#44;0.057763327&#44;0.06836419&#44;-0.12669459&#44;-0.29591&#44;-0.3165992&#44;-0.24508218&#44;-0.38491094&#44;-0.21221805&#44;0.036103904&#44;0.019153569&#44;0.14388251&#44;-0.0377556&#44;-0.035323378&#44;-0.06883999&#44;-0.043445915&#44;-0.041686647&#44;0.023585027&#44;-0.1399101&#44;-0.1010129&#44;-0.18053892&#44;-0.029968912&#44;-0.17923358&#44;0.22831741&#44;0.08030359&#44;-0.16442463&#44;0.1414377&#44;0.13718733&#44;-0.03144061&#44;-0.0054521905&#93;</td>
        <td>q: 产假的具体内容是什么？</td>
        <td>&#123;&#34;a&#34;:&#34;符合国家计划生育条例规定的女员工依法享受相应产假，在产前 15 天开始计入产假。配偶生育的给予男员工相应陪产假。2、孕期女员工在工作时间内进行产前检查必须提供政府计生部门出具的准生证明原件及当地区级或二级以上医院开具的产检证明，则视同出勤，产检假期如下：怀孕第 1-6 个月，每个月可享受 1 天假期;怀孕第 7-8 个月，每月可享受 2 天假期;怀孕 9 个月以上，每月可享受 4 天假期，但其中 2 天已包括在预产假中。产检假提前至少 1 天在钉钉申请。3、女员工申请产假前须提供准生证明原件，提前至少 3 天在钉钉申请产假，人力资源部审核确认并存档。在产后 90 天内将婴儿出生证明原件及其他证明原件交至人力资源部，以便核实具体产假时间及办理相关生育费用报销手续。4、女员工在婴儿 1 周岁以前，公司给予其每天 1 小时的哺乳时间，对于生育多胞胎的，每多 1 个婴儿每天增加 1 小时哺乳时间。具体时间须提前与直属上级和部门负责人协商，报人力资源部审核确认并存档。5、男员工申请陪产假必须提供婴儿出生证明原件和结婚证原件，自婴儿出生后 2 个月内休完，逾期视其自动放弃休假，且公司不予任何形式补偿。试用期男员工陪产假需转正后方可申请。6、产假、产检假、陪产假、流产、男女员工节育假期按国家规定必须为连续不间断计算，均包括法定节假日和双休日。产假及相关衍生假期间按照不低于当地最低工资标准发放工资，待生育津贴报销后补齐差额，不计发其他奖金福利补助。7、违反国家计划生育条例规定的女员工不享受产假与产检假期间的工资待遇及其他奖金福利补助，期间按事假规定处理，且自行承担生育等费用。&#34;&#44;&#34;fileName&#34;:&#34;dataset/dataset-playground/v1/qa.csv&#34;&#44;&#34;lineNumber&#34;:16&#125;</td>
        <td>1f8d07a4-1ffe-4c93-b2eb-94982e94a775</td>
        <td>06a80369-75ad-4269-86bc-7e7a7da2b356</td>
    </tr>
</table>

### CODE REVIEW

**Comment by Abirdcfly** [comment]
**Do not merge**

As you can see here the database for the `pgvector` uses `admin`, but it looks like the database for the `pg` configuration is `arcadia`, need to figure out what went wrong and why the `arcadia` database wasn't created, but the `admin` database was.

File: .github/workflows/pgvector_image_build.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  pull_request:
+    branches: [main]
+  push:
+    branches: [main]
+    paths:
+      - 'deploy/pgvector/Dockerfile'
+  workflow_dispatch:
+
+env:
+  TAG: 16.1.0-debian-11-r18-pgvector-v0.5.1
```
Comment (bjwswang) [inline .github/workflows/pgvector_image_build.yml:13]: Why this tag?

File: api/base/v1alpha1/vectorstore_types.go [hunk 2]
```diff
@@ -29,13 +29,28 @@ type VectorStoreSpec struct {
...
 // Chroma defines the configuration of Chroma
 type Chroma struct {
 	DistanceFunction chromago.DistanceFunction `json:"distanceFunction,omitempty"`
 }
 
+type PGVector struct {
+	// PreDeleteCollection defines if the collection should be deleted before creating.
+	PreDeleteCollection bool `json:"preDeleteCollection,omitempty"`
+	// CollectionName defines the name of the collection
+	CollectionName string `json:"collectionName,omitempty"`
```
Comment (bjwswang) [inline api/base/v1alpha1/vectorstore_types.go:45]: What are the differences between  `CollectionName  `  and  `CollectionTableName `  ?

Comment (Abirdcfly) [reply api/base/v1alpha1/vectorstore_types.go:45]: https://github.com/langchain-ai/langchain/blob/85a4594ed7dbde9d257e230d5d789b68c42421cf/libs/community/langchain_community/vectorstores/pgvector.py#L154C27-L154C27

These are configurables introduced by langchain python, which langchain-go replicates, and is just exposing them for advanced users to use here.
Normal users don't need to configure it, just use the default value is enough.

**Review by bjwswang — APPROVED** [review]

**Pull request merged**